lib/Sema/CSSimplify.cpp

```diff
@@ -1771,17 +1832,33 @@ static bool fixMissingArguments(ConstraintSystem &cs, ASTNode anchor,
         cs.createTypeVariable(argLoc, TVO_CanBindToNoEscape)));
   }
 
-  SmallVector<std::pair<unsigned, AnyFunctionType::Param>, 4> synthesizedArgs;
+  SmallVector<SynthesizedArg, 4> synthesizedArgs;
   synthesizedArgs.reserve(numMissing);
   for (unsigned i = args.size() - numMissing, n = args.size(); i != n; ++i) {
-    synthesizedArgs.push_back(std::make_pair(i, args[i]));
+    synthesizedArgs.push_back(SynthesizedArg{i, None, args[i]});
   }
 
-  auto *fix = AddMissingArguments::create(cs, synthesizedArgs,
-                                          cs.getConstraintLocator(locator));
+  bool shouldFix = true;
 
-  if (cs.recordFix(fix))
-    return true;
+  // Treat missing anonymous arguments as valid in closures containing the
+  // code completion location, since they may have just not been written yet.
+  if (cs.isForCodeCompletion()) {
+    if (auto *closure = getAsExpr<ClosureExpr>(anchor)) {
+      SourceManager &SM = closure->getASTContext().SourceMgr;
+      SourceRange range = closure->getSourceRange();
+      if (range.isValid() && SM.rangeContainsCodeCompletionLoc(range)) {
+        shouldFix = !closure->hasAnonymousClosureVars() &&
+          (!args.empty() || closure->getInLoc().isValid());
```

Comment:
Maybe an early return here in reverse situation instead?

Reply:
Would missing the single argument tuple handling below (need to expand a few lines) be a problem?

Reply:
Yeah, I think that should be okay.